source/templates.tex

```diff
@@ -3736,10 +3735,13 @@
 parameter (i.e., has the same constant value or the same type
 as the template parameter) can be used in place of that
 template parameter in a reference to the current
-instantiation. In the case of a non-type template argument,
+instantiation.
+\begin{note}
+In the case of a non-type template argument,
```

Comment:
We've discussed this recently for core issue 2307. Why is it safe to turn the normative text into a note here?

Reply:
`N` is not equivalent to, say, `(N)` per [temp.over.link]p5.

I'm happy to make a different change if you'd prefer. Perhaps something more in this direction:

> A template argument that is equivalent to a template parameter (i.e., has the same type as the template parameter or, *for a non-type template argument, is an expression that names the template parameter but not as a subexpression*) can be used in place of that template parameter in a reference to the current instantiation.

... but that's pretty awkward. Or perhaps:

> A template argument that is equivalent to a template parameter (i.e., has the same constant value or the same type as the template parameter) can be used in place of that template parameter in a reference to the current instantiation. In the case of a non-type template argument, this *only applies if the argument was* given the value of the template parameter and not an expression in which the template parameter appears as a subexpression.

Reply:
Well, we don't really mean the "equivalent" from [temp.over.link] here: the example claims that my_I and I are "equivalent", but they are not the same per the ODR (i.e. token-by-token identical). So, we need the normative text to explain the different notion of "equivalent". [The very fact we're even discussing this makes me uneasy about such a large-ish last-minute change.]

That said, your second proposal above seems to be least invasive, so I'd prefer that.

Reply:
Done.